docs/tutorial/tutorial.md

```diff
@@ -132,44 +132,72 @@ After: You should see a number in each square in the rendered output.
 
 ### An Interactive Component
 
-Let's make the Square component fill in an "X" when you click it. Try changing the button tag returned in the `render()` function of the `Square` class to:
+Let's make the Square component fill in an "X" when you click it. Try changing the button tag returned in the `render()` function of the Square like this:
 
-```html
-<button className="square" onClick={() => alert('click')}>
-  {/* TODO */}
-</button>
+```javascript{4-6}
+class Square extends React.Component {
+  render() {
+    return (
+      <button className="square" onClick={() => alert('click')}>
+        {/* TODO */}
+      </button>
+    );
+  }
+}
 ```
 
 This uses the new JavaScript arrow function syntax. If you click on a square now, you should get an alert in your browser.
 
+Note that we're passing a function as the `onClick` prop. Doing `onClick={alert('click')}` would alert immediately instead of when the button is clicked.
+
 React components can have state by setting `this.state` in the constructor, which should be considered private to the component. Let's store the current value of the square in state, and change it when the square is clicked. First, add a constructor to the class to initialize the state:
 
-```javascript
+```javascript{2-7}
 class Square extends React.Component {
   constructor() {
     super();
     this.state = {
       value: null,
     };
   }
-  // ...
+  render() {
+    return (
+      <button className="square" onClick={() => alert('click')}>
+        {/* TODO */}
+      </button>
+    );
+  }
 }
 ```
 
 In JavaScript classes, you need to explicitly call `super();` when defining the constructor of a subclass.
 
 Now change the `render` method to display `this.state.value` instead of `this.props.value`, and change the event handler to  be `() => this.setState({value: 'X'})` instead of the alert:
 
-```javascript
-<button className="square" onClick={() => this.setState({value: 'X'})}>
-  {this.state.value}
-</button>
+```javascript{10-12}
+class Square extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      value: null,
+    };
+  }
+  render() {
+    return (
+      <button className="square" onClick={() => this.setState({value: 'X'})}>
+        {this.state.value}
+      </button>
+    );
+  }
+}
 ```
 
 Whenever `this.setState` is called, an update to the component is scheduled, causing React to merge in the passed state update and rerender the component along with its descendants. When the component rerenders, `this.state.value` will be `'X'` so you'll see an X in the grid.
 
 If you click on any square, an X should show up in it.
 
+<a href="https://codepen.io/brigand/pen/XRjPRg?editors=0010" target="_blank">View the current code</a>.
```

Comment:
"If you click on any square, an X should show up in it."

I don't think the linked CodePen does this. I see an alert in it.